nft/nft_lib/src/main.sw

```diff
@@ -0,0 +1,139 @@
+library nft_lib;
+
+use std::{
+    address::Address,
+    assert::require,
+    constants::NATIVE_ASSET_ID,
+    chain::auth::{AuthError, Sender, msg_sender},
+    result::*,
+    revert::revert,
+};
+
+enum Error {
+    AddressCannotBeZero: (),
+    NFTNotInitalized: (),
+    OwnerAlreadySet: (),
+    OwnerCannotBeApproved: (),
+    SenderNotOwner: (), 
+}
+
+// TODO: Add mappings when supported for token ids
+
+struct Data {
+    owner: Address,
+    approved: Address,
+}
+
+storage {
+    data: Data,
+    state: u64,
+}
+
+/// Sets the inital owner for the NFT
+///
+/// # Panics
+///
+/// The function will panic when:
+/// - The NFT has already been initalized
+/// - The new owner is the 0 address
+pub fn mint_to_address(new_owner: Address) -> bool {
+    require(storage.state == 0, Error::OwnerAlreadySet);
+    require(new_owner.value != NATIVE_ASSET_ID, Error::AddressCannotBeZero);
+
+    storage.data = Data {
+        owner: new_owner, approved: ~Address::from(NATIVE_ASSET_ID)
```

Comment:
Isn't NATIVE_ASSET_ID alreadt an address?  Do we need the ~Address::from part here?

Reply:
It is not an address, it's a [constant](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/constants.sw#L3)

Reply:
The Sway compiler can't yet do constant evaluation, so constants (such as `NATIVE_ASSET_ID`) must be primitives (such as `b256` in the case of `NATIVE_ASSET_ID`).